WordPress/Classes/Services/PostService+RefreshStatus.swift

```diff
@@ -9,22 +9,24 @@ extension PostService {
     ///
     func refreshPostStatus(onCompletion: (() -> Void)? = nil, onError: ((Error) -> Void)? = nil) {
         self.managedObjectContext.perform {
-            let fetch = NSFetchRequest<Post>(entityName: Post.classNameWithoutNamespaces())
+            let request = NSBatchUpdateRequest(entityName: Post.classNameWithoutNamespaces())
             let pushingPredicate = NSPredicate(format: "remoteStatusNumber = %@", NSNumber(value: AbstractPostRemoteStatus.pushing.rawValue))
             let processingPredicate = NSPredicate(format: "remoteStatusNumber = %@", NSNumber(value: AbstractPostRemoteStatus.pushingMedia.rawValue))
-            fetch.predicate = NSCompoundPredicate(orPredicateWithSubpredicates: [pushingPredicate, processingPredicate])
+            let pushingOrProcessingPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: [pushingPredicate, processingPredicate])
+            let notFailedPredicate = NSPredicate(format: "remoteStatusNumber != %@ AND NOT (postID != nil AND postID > 0)", NSNumber(value: AbstractPostRemoteStatus.failed.rawValue))
+            request.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [pushingOrProcessingPredicate, notFailedPredicate])
+            request.propertiesToUpdate = [
+                "remoteStatusNumber": NSNumber(value: AbstractPostRemoteStatus.failed.rawValue),
+                "status": NSString(string: BasePost.Status.draft.rawValue),
+                "dateModified": NSDate()
+            ]
             do {
-                let postsPushing = try self.managedObjectContext.fetch(fetch)
-                for post in postsPushing {
-                    self.markAsFailedAndDraftIfNeeded(post: post)
```

Comment:
Is there nothing in the logic of this method that needs to be handled in a refresh state?

Reply:
the logic from this method contained a guard for a post not being failed and not having a remote, which I've represented as `notFailedPredicate` on line 16